protocol/integration/protocol_test.go

```diff
@@ -1815,10 +1815,11 @@ func TestConsumerProviderWithProviderSideCache(t *testing.T) {
 	// Get latest for sanity check
 	sendMessage("status", []string{})
 
-	for i := 0; i < 5; i++ {
-		// Get block
-		sendMessage("block", []string{"1000"})
-	}
+	// Get block, this should be cached for next time
+	sendMessage("block", []string{"1000"})
+
+	// Get block again, this time it should be from cache
+	sendMessage("block", []string{"1000"})
```

Comment:
it might fail because a routine will add to cache. so race here is gonna be constantly happening. verify its in cache before checking the result. 

do some thing like
for {
    if cache.inputFound("1000") { break} 
}

Reply:
Fixed in aacaa362ff440265da81f630258abf9cdec22b73